tests/spark_autologging/test_pyspark_ml_autologging.py

```diff
@@ -0,0 +1,238 @@
+)
+
+from tests.spark_autologging.utils import spark_session  # pylint: disable=unused-import
+import pyspark
+from pyspark.ml.linalg import Vectors
+from pyspark.ml.regression import LinearRegression, LinearRegressionModel
+from mlflow.pyspark.ml import (
+    _should_log_model,
+    _get_instance_param_map,
+    _get_warning_msg_for_skip_log_model,
+    _get_warning_msg_for_fit_call_with_a_list_of_params,
+)
+
+MODEL_DIR = "model"
+MLFLOW_PARENT_RUN_ID = "mlflow.parentRunId"
+
+
+@pytest.fixture(scope="module")
+def dataset_binomial(spark_session):
+    return spark_session.createDataFrame(
+        [(1.0, Vectors.dense(1.0)), (0.0, Vectors.sparse(1, [], []))] * 100, ["label", "features"]
+    )
+
+
+@pytest.fixture(scope="module")
+def dataset_multinomial(spark_session):
+    return spark_session.createDataFrame(
+        [(1.0, Vectors.dense(1.0)), (0.0, Vectors.sparse(1, [], [])), (2.0, Vectors.dense(0.5))]
+        * 100,
+        ["label", "features"],
+    )
+
+
+def truncate_param_dict(d):
+    return truncate_dict(d, MAX_ENTITY_KEY_LENGTH, MAX_PARAM_VAL_LENGTH)
+
+
+def stringify_dict_values(d):
+    return {k: str(v) for k, v in d.items()}
+
+
+def get_expected_class_tags(estimator):
+    return {
+        "estimator_name": estimator.__class__.__name__,
+        "estimator_class": estimator.__class__.__module__ + "." + estimator.__class__.__name__,
+    }
+
+
+def get_run_data(run_id):
+    client = mlflow.tracking.MlflowClient()
+    data = client.get_run(run_id).data
+    # Ignore tags mlflow logs by default (e.g. "mlflow.user")
+    tags = {k: v for k, v in data.tags.items() if not k.startswith("mlflow.")}
+    artifacts = [f.path for f in client.list_artifacts(run_id)]
+
+    RunData = namedtuple("RunData", ["params", "metrics", "tags", "artifacts"])
+    return RunData(data.params, data.metrics, tags, artifacts)
+
+
+def load_model_by_run_id(run_id):
+    return mlflow.spark.load_model("runs:/{}/{}".format(run_id, MODEL_DIR))
+
+
+def get_dataset_binomial(spark_session):
+    return spark_session.createDataFrame(
+        [(1.0, Vectors.dense(1.0)), (0.0, Vectors.sparse(1, [], []))] * 100, ["label", "features"]
+    )
+
+
+def test_basic_estimator(dataset_binomial):
+    mlflow.pyspark.ml.autolog()
+    lr = LinearRegression()
+    with mlflow.start_run() as run:
+        lr_model = lr.fit(dataset_binomial)
+    run_id = run.info.run_id
+    run_data = get_run_data(run_id)
+    assert run_data.params == truncate_param_dict(
+        stringify_dict_values(_get_instance_param_map(lr))
+    )
+    assert run_data.tags == get_expected_class_tags(lr)
+    assert MODEL_DIR in run_data.artifacts
+    loaded_model = load_model_by_run_id(run_id)
+    assert loaded_model.stages[0].uid == lr_model.uid
+
+
+@pytest.mark.skipif(
+    LooseVersion(pyspark.__version__) < LooseVersion("3.1"),
+    reason="This test fails on supported versions of sklearn",
+)
+def test_models_in_allowlist_exist(spark_session):  # pylint: disable=unused-argument
```

Comment:
```suggestion
def test_models_in_allowlist_exist():
```

Do we need `spark_session` here?

Reply:
yes. see above comment